public_html/wp-content/mu-plugins/blocks/assets/src/speakers/speakers-select.js

```diff
@@ -151,9 +159,12 @@ class SpeakersSelect extends Component {
 					);
 				} }
 				onChange={ ( selectedOptions ) => {
+					// this seems like it's big enough that it'd be more readable if it were modularized into a named function
+					// even the smaller ones above feel like they could be too. if it's not a 1-liner then it makes the element harder to scan quickly, and mixes logic with markup
```

Comment:
> if it's not a 1-liner then it makes the element harder to scan quickly, and mixes logic with markup

Good point about mixing logic with markup, although that's the whole problem with React in general, right? :D

I'm not sure about scanability, though. If you have to scroll back and forth to different parts of the file to figure out what the named callback does, is it really more readable?

(I'm just being pedantic. I think I have done it both ways, even within the codebase of this block. We should probably pick one pattern and stick with it)

Reply:
haha, that's true :)

it's seems like there are cases where it helps to keep clear boundaries, though, and to me this feels like one of them. i don't feel strongly, though.

having to scroll to find out what the callback does is a valid point. hopefully that would be somewhat mitigated by a descriptive function name, and `cmd+b` and `cmd+alt+left` in PhpStorm, but it would still add that extra work in some cases.

it might just come down to personal preference for where the lines get drawn, and if that's the case i'm fine leaving it as a case-by-case judgement call.